File: resources/Torus World/scripts/Director.js
ZOrderStage = itchy.ZOrderStage;
WrappedCollisionStrategy = itchy.collision.WrappedCollisionStrategy;
WrappedStageView = itchy.WrappedStageView;
StageView = itchy.StageView;

Director = Class({

    Extends : DirectorScript,

    init : function() {
        this.score = 0;
        this.lives = 0;
    },

    onStarted : function() {
        // Don't create default stages and views, because we want to use a
        // special WrappedStageView
        stdout.println("Creating custom stages and views. " + game);

        var screenRect = Rect(0, 0, game.getWidth(), game.getHeight());

        this.mainStage = ZOrderStage("main");
        game.getStages().add(this.mainStage);

        // We need a separate view for the "lives" in the top left, because we
        // don't want those to
        // wrap round to the bottom of the screen.
        this.hudStage = ZOrderStage("hud");
        game.getStages().add(this.hudStage);

        this.mainView = WrappedStageView(screenRect, this.mainStage);
        this.mainView.wrap(game.getHeight(), game.getWidth(), 0, 0);
        game.getGameViews().add(this.mainView);

        this.hudView = StageView(screenRect, this.hudStage);
        game.getGameViews().add(this.hudView);

        this.hudView.enableMouseListener(game);
        this.collisionStrategy = WrappedCollisionStrategy(this.mainView);
    },

    startScene : function(sceneName) {
        stdout.println("Starting scene : " + sceneName);
        if (game.pause.isPaused()) {
            game.pause.unpause();
        }
        if (sceneName == "menu" && game.getSceneName() == "menu") {
            return true;
        }
        var transition = itchy.extras.SceneTransition.fade();
        if (game.getSceneName() == "menu") {
            transition = itchy.extras.SceneTransition.slideRight();
        }
        if (sceneName == "menu") {
            transition = itchy.extras.SceneTransition.slideLeft();
        }
        return new itchy.extras.SceneTransition(transition).transition(sceneName);
    },

    onMessage : function(message) {
        if (message == "start") {
            this.startGame("1");
        }

        if (message == "continue") {
            this.startGame(game.getSceneName());
        }
    },

    startGame : function(sceneName) {
        if (!sceneName) {
            sceneName = "1";
        }
        this.score = 0;
        this.lives = 3;
        this.startScene(sceneName);
    },

    addPoints : function(points) {
        this.score += points;
    },

    onWindowEvent : function(event) {
        if (event.lostMouseFocus()) {
            game.mouse.showRegularMousePointer(true);
            return true;
        } else if (event.gainedMouseFocus()) {
            this.showFancyMouse();
            return true;
        }
        return false;
    },

    showFancyMouse : function() {
        var mousePointer = new itchy.extras.SimpleMousePointer("mouse");
        game.mouse.setMousePointer(mousePointer);
        new itchy.role.Explosion(mousePointer.getActor())
            .dependent().forever().follow().offset(40, -33)
            .projectilesPerTick(1).spread(-20,-80).distance(10).randomSpread()
            .speed(1, 2, 0, 0).fade(3).eventName("spark").createActor();
        new itchy.role.OnionSkin(mousePointer.getActor()).alpha(128).fade(3).every(1).createActor();
    }

});

File: resources/Space-Miner/scripts/director.py
from uk.co.nickthecoder.itchy import Itchy
from uk.co.nickthecoder.itchy import AbstractDirector
from uk.co.nickthecoder.itchy import StageView
from uk.co.nickthecoder.itchy import WrappedStageView
from uk.co.nickthecoder.itchy import ZOrderStage
from uk.co.nickthecoder.itchy.collision import WrappedCollisionStrategy
from uk.co.nickthecoder.itchy.extras import SceneTransition
from uk.co.nickthecoder.itchy.extras import SimpleMousePointer
from uk.co.nickthecoder.itchy.role import Explosion
from uk.co.nickthecoder.itchy.role import OnionSkin
from uk.co.nickthecoder.itchy.util import ClassName

from uk.co.nickthecoder.jame import Rect

from java.util import ArrayList

class Director(AbstractDirector) :

    def __init__(self) :
        self.score = 0
        self.lives = 3

    def onStarted(self) :
        self.socre = 0
        self.lives = 3
        
        # Don't create default stages and views, because we want to use a special WrappedStageView
        print "Creating custom stages and views"

        screenRect = Rect(0, 0, self.game.getWidth(), self.game.getHeight())

        self.mainStage = ZOrderStage("main")
        self.game.getStages().add(self.mainStage)

        # We need a separate view for the "lives" in the top left, because we don't want those to
        # wrap round to the bottom of the screen.
        self.hudStage = ZOrderStage("hud")
        self.game.getStages().add(self.hudStage)

        self.mainView = WrappedStageView(screenRect, self.mainStage)
        self.mainView.wrap( self.game.getHeight(), self.game.getWidth(), 0, 0 )
        self.game.getGameViews().add(self.mainView)

        self.hudView = StageView(screenRect, self.hudStage)
        self.game.getGameViews().add(self.hudView)

        self.hudView.enableMouseListener(self.game)
        self.collisionStrategy = WrappedCollisionStrategy( self.mainView )

    def tick(self) :
        pass
      
    def startScene(self, sceneName) :
        print "Starting scene :", sceneName
        
        if self.getGame().pause.isPaused() :
            self.getGame().pause.unpause()
            
        if sceneName == "menu" and self.getGame().getSceneName() == "menu" :
            return True;
        
        transition = SceneTransition.fade();
        if self.getGame().getSceneName() == "menu" :
            transition = SceneTransition.slideRight()

        if sceneName == "menu" :
            transition = SceneTransition.slideLeft()
            self.score = 0
            self.lives = 3

        return SceneTransition(transition).transition(sceneName)
    

    def onMessage(self, message) :
        if message == "start" :
            self.startGame("1")
        
        if message == "continue" :
            self.startGame(self.getGame().getSceneName())
    
    def addPoints(self, points ) :
        self.score += points;

    def onWindowEvent(self, event) :
        if event.lostMouseFocus() :
            self.getGame().mouse.showRegularMousePointer( True )
            return True
        elif event.gainedMouseFocus() :
            self.showFancyMouse()
            return True
        return False

    def showFancyMouse(self) :
        mousePointer = SimpleMousePointer("mouse");
        self.getGame().mouse.setMousePointer( mousePointer );
        
        Explosion(mousePointer.getActor()).dependent().forever().follow().offset(40,-33).projectilesPerTick(1) \
            .spread(-20,-80).distance(10).randomSpread().speed(1,2,0,0).fade(3).eventName("spark") \
            .createActor()
        OnionSkin(mousePointer.getActor()).alpha(128).fade(3).every(1).createActor()

    # Boiler plate code - no need to change this
    def getClassName(self):
        return ClassName( CostumeProperties, self.__module__ + ".py" )

 
